src/modules/SdnDiag.Health.psm1

```diff
@@ -2,1467 +2,1638 @@
+        Checks if the provided service role is an SDN cluster service role
 
-            "DNS name for {0} resolves to {1} instead of {2}" -f $nbApiName, $dnsResult[0].IPAddress, $expectedIPAddress | Trace-Output -Level:Warning
-            return $sdnHealthObject
-        }
+        .PARAMETER ServiceName
+        The name of the service role to check
+    #>
 
-        return $sdnHealthObject
-    }
-    catch {
-        $_ | Trace-Exception
-        $_ | Write-Error
-    }
+    param([string] $ServiceName)
+
+    # Define the list of valid service roles
+    $validServiceRoles = @(
+        "ApiService",
+        "ControllerService",
+        "FirewallService",
+        "FnmService",
+        "GatewayManager",
+        "ServiceInsertion",
+        "VSwitchService"
+    )
+
+    # Check if the provided service role name is in the list
+    return $validServiceRoles -contains $ServiceName
 }
+function IsSdnService {
 
-function Test-NetworkControllerCertCredential {
     <#
-    .SYNOPSIS
-        Query the NC Cert credential used to connect to SDN Servers, ensure cert exist.
+        .SYNOPSIS
+        Checks if the provided service name is an SDN agent service
+
+        .PARAMETER serviceName
+        The name of the service to check
     #>
 
-    [CmdletBinding(DefaultParameterSetName = 'RestCredential')]
-    param (
-        [Parameter(Mandatory = $true)]
-        [SdnFabricEnvObject]$SdnEnvironmentObject,
+    param([string] $serviceName)
 
-        [Parameter(Mandatory = $false)]
-        [System.Management.Automation.PSCredential]
-        [System.Management.Automation.Credential()]
-        $Credential = [System.Management.Automation.PSCredential]::Empty,
+    return $serviceName -in @( "NCHostAgent", "SlbHostAgent")
+}
 
-        [Parameter(Mandatory = $false, ParameterSetName = 'RestCredential')]
-        [System.Management.Automation.PSCredential]
-        [System.Management.Automation.Credential()]
-        $NcRestCredential = [System.Management.Automation.PSCredential]::Empty,
+function IsCurrentNodeClusterOwner {
+    <#
+        .SYNOPSIS
+        Checks if the current node is the owner of the cluster
 
-        [Parameter(Mandatory = $true, ParameterSetName = 'RestCertificate')]
-        [X509Certificate]$NcRestCertificate
-    )
+        .NOTES
+        This function is used to determine if the current node is the owner of the cluster. This is used to determine if the current node is the primary node in a cluster.
+    #>
 
-    $ncRestParams = @{
-        NcUri = $SdnEnvironmentObject.NcUrl
-    }
-    switch ($PSCmdlet.ParameterSetName) {
-        'RestCertificate' {
-            $ncRestParams.Add('NcRestCertificate', $NcRestCertificate)
-        }
-        'RestCredential' {
-            $ncRestParams.Add('NcRestCredential', $NcRestCredential)
-        }
+    $activeNode = Get-ClusterResource  | ? { $_.OwnerGroup -eq "Cluster Group" -and $_.ResourceType -eq "IP Address" -and $_.Name -eq "Cluster IP Address" }
```

Comment:
We should add an `-ErrorAction Ignore` if this cmdlet fails since you just checking if $null

Reply:
done